### PR TITLE
feat: add detailed error messages for various HTTP status codes

### PR DESCRIPTION
**Pull request by marliessophie**

### CODE REVIEW

File: langfuse-core/src/index.ts [hunk 1]
```diff
@@ -87,8 +87,75 @@ class LangfuseFetchNetworkError extends Error {
...
+// Error messages
+const updatePromptResponse = `Make sure to keep your SDK updated, refer to ${NPM_PACKAGE_URL} for details.`;
+const defaultErrorResponse = `Unexpected error occurred. Please check your request and contact support: ${SUPPORT_URL}.`;
+
+// Error response map
+const errorResponseByCode = new Map<number, string>([
+  // Internal error category: 5xx errors, 404 error
+  [500, `Internal server error occurred. Please contact support: ${SUPPORT_URL}`],
+  [501, `Not implemented. Please check your request and contact support: ${SUPPORT_URL}.`],
+  [502, `Bad gateway. Please try again later and contact support: ${SUPPORT_URL}.`],
```
Comment (maxdeichmann) [inline langfuse-core/src/index.ts:118]: "try again later" not the perfect message i think. I would say something along the lines:

Bad gateway. This should not happen and we are monitoring this closely. For help, please contact support: ${SUPPORT_URL}.

Comment (maxdeichmann) [reply langfuse-core/src/index.ts:118]: Same for the messages below

File: langfuse-core/src/index.ts [hunk 2]
```diff
@@ -87,8 +87,75 @@ class LangfuseFetchNetworkError extends Error {
...
+  [503, `Service unavailable. Please try again later and contact support if the error persists: ${SUPPORT_URL}.`],
+  [504, "Gateway timeout. Please try again later and contact support: ${SUPPORT_URL}."],
+  [
+    404,
+    `Internal error occurred. Likely caused by race condition, please escalate to support if seen in high volume: ${SUPPORT_URL}`,
```
Comment (maxdeichmann) [inline langfuse-core/src/index.ts:123]: If we write this, people think that we do not have our stack under control and dont account for concurrency in our architecture. I would handle this in the ingestion pipeline similar to a 5xx error.

File: langfuse-core/src/index.ts [hunk 3]
```diff
@@ -87,8 +87,75 @@ class LangfuseFetchNetworkError extends Error {
...
+  ],
+
+  // Client error category: 4xx errors, excluding 404
+  [
+    400,
+    `Bad request. Please check your request for any missing or incorrect parameters. Refer to our API docs: ${API_DOCS_URL} for details.`,
+  ],
+  [401, `Unauthorized. Please check your public/private host settings.`],
```
Comment (maxdeichmann) [inline langfuse-core/src/index.ts:131]: Can we add a link here pointing to how to set up the SDK with the correct configs?

File: langfuse-core/src/index.ts [hunk 4]
```diff
@@ -87,8 +87,75 @@ class LangfuseFetchNetworkError extends Error {
...
+  [403, `Forbidden. Please check your access control settings. Refer to our RBAC docs: ${RBAC_DOCS_URL} for details.`],
+  [
+    429,
+    `Rate limit exceeded. Please try again later. For more information on rate limits please see: ${RATE_LIMITS_URL}`,
```
Comment (maxdeichmann) [inline langfuse-core/src/index.ts:135]: I would also remove "Please try again later."

File: langfuse-core/src/index.ts (hunk 2 above)
Comment (marliessophie) [reply langfuse-core/src/index.ts:123]: thanks! So rather something along the lines of "This should not happen and we are monitoring this closely. For help, please contact support: ${SUPPORT_URL}."

**Pull request merged**